chore: update changelog and versioning for 2.3.0 (#721)
awslabs/aws-advanced-jdbc-wrapper: CHANGELOG.md

- Native telemetry support ([PR #617](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/617)).
- Documentation on known limitations with global databases ([PR #695](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/695)).

### :bug: Fixed
- Continue monitoring if unhandled Exception is thrown ([PR #676](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/676)).
- Password properties are now masked in logs ([PR #701](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/701)).
- Issue when getting a connection for a closed statement ([PR #682](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/682)).
- Maven coordinates in README ([PR #681](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/681)).
- Update topology for specific methods ([PR #683](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/683)).

### :crab: Changed
- Added buffer to IAM token expiry and moved token expiry time creation ([PR #706](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/706)).

> - Documentation on known limitations with Blue/Green deployments. ([PR #680](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/680)).

Reviewer: ```suggestion
- Documentation on known limitations with Blue/Green deployments ([PR #680](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/680)).
```

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/#semantic-versioning-200).

## [2.3.0] - 2023-11-23
### :magic_wand: Added
- Endpoint override for secrets manager plugin ([PR #707](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/707)).
- Allow users to set up a lambda to initialize new connections ([PR #705](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/705)).

> - Introduced Dialect.prepareConnectProperties ([PR #704](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/704)).

Reviewer 2: ```suggestion
- Introduced `Dialect.prepareConnectProperties` ([PR #704](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/704)).
```

> - Introduced Dialect.prepareConnectProperties ([PR #704](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/704)).

Reviewer: ```suggestion
- Introduced Dialect.prepareConnectProperties to allow dialect classes to modify connection properties when opening a new connection ([PR #704](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/704)).
```

> - Native telemetry support in the AWS JDBC Driver ([PR #617](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/617)).

Reviewer 2: Didn't add this in line, but should we consider "Native Open Telemetry support"?

```suggestion
- Native telemetry support ([PR #617](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/617)).
```

Author: Telemetry is the feature we are adding, and OpenTelemetry is a standard that we are using to enable Telemetry. The documentation specifies it - we are adding native support for instrumentation based on the OpenTelemetry standard - not sure if we should mention OpenTelemetry here

Reviewer 2: That's fine, let's keep it with your original proposal.

> - Native telemetry support in the AWS JDBC Driver ([PR #617](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/617)).

Reviewer: ```suggestion
- Native telemetry support in the AWS JDBC Driver ([PR #617](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/617)).
- Documentation on known limitations with global databases ([PR #695](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/695)).
```

### :bug: Fixed
- Continue monitoring if unhandled Exception is thrown ([PR #676](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/676)).
- Password properties are now masked in logs ([PR #701](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/701)).
- Issue when getting a connection for a closed statement ([PR #682](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/682)).

> - Maven coordinates in README ([PR #681](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/681)).

Reviewer: ```suggestion
- Maven coordinates in README ([PR #681](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/681)).
- Update topology for specific methods ([PR #683](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/683)).
```

### :crab: Changed

> - Added buffer to iam token expiry and create token expiry time earlier ([PR #706](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/706)).

Reviewer 2: Could we clarify earlier here? I've suggested removing it since it's not too clear.

```suggestion
- Added buffer to IAM token expiry and create token expiry time ([PR #706](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/706)).
```

> - Updated AWS SDK dependency versions ([PR #708](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/708)).

Reviewer: We can probably remove this line since it's just updating dependency versioning

> - Documentation w.r.t known limitations with global databases ([PR #695](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/695)).
> - Topology for specific methods ([PR #683](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/683)).
> - Clarified documentation for limitations with Blue/Green deployments. See [Known Limitations](https://github.com/awslabs/aws-advanced-jdbc-wrapper/blob/main/docs/KnownLimitations.md#amazon-rds-bluegreen-deployments).

Reviewer: ```suggestion
- Documentation on known limitations with Blue/Green deployments. ([PR #680](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/680)).
```

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/#semantic-versioning-200).

## [2.3.0] - 2023-11-23
### :magic_wand: Added

> - Endpoint override for secrets manager plugin ([PR #707](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/707)).

Reviewer 2: ```suggestion
- Endpoint override for the AWS Secrets Manager plugin ([PR #707](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/707)).
```

- Support for AWS RDS Multi-AZ DB Cluster, MySQL and Postgres ([PR #690](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/690)).
- Allow user to set up a lambda to initialize new connections ([PR #705](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/705)).
- Introduced Dialect.prepareConnectProperties ([PR #704](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/704)).

> - Added native telemetry support in the AWS JDBC Driver ([PR #617](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/617)).

Reviewer: ```suggestion
- Native telemetry support in the AWS JDBC Driver ([PR #617](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/617)).
```

> - Added native telemetry support in the AWS JDBC Driver ([PR #617](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/617)).

Reviewer 2: ```suggestion
- Added native telemetry support ([PR #617](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/617)).
```
